### PR TITLE
Updates Gamma Ship

### PR DESCRIPTION
**Pull request by Kyep**
Currently, the Cyberiad's Gamma armory spends most of its time looking like this:

This is effectively an empty room for jaunting vampires, wizards, and similar to hide in. It looks weird, and encourages people to hide in there, safe in the knowledge that it has breathable air, but crew can't get in.

When gamma is called, the interior is simply replaced by this:

This makes little sense, for several reasons. First, the whole room is filled as if it had always been there. But it hasn't been. Its effectively a mass-teleport trying to pass itself off as a sealed room. Second, some of the contents don't make sense. Desk charger? Weird dual-floor layout? Glass panes? Etc. It could be improved.

This PR starts off by having Gamma as effectively a docking port in space:

new_gamma_unmoved_mech

When Gamma is deployed, it moves a Gamma pod into place, to provide Gamma supplies:

new_gamma_moved_mech

This has several advantages over the old layout:
- Wizards, Vampires, etc can no longer hide in empty 'gamma room' before gamma is called. That section is now simply space. Even if they manage to get in, there is a window, so they might be seen. The gamma armory no longer supports antags being cheesy with teleports.
- The movement of the gamma armory is no longer a mysterious bluespace teleport. Now, it acts like a shuttle pod that docks with the station. It isn't *really* a shuttle, it still keeps the existing move_gamma_ship proc, but at least it looks like something physically possible to players now. It looks like a shuttle/pod.
- The contents of the room have been updated to modern NT standards. The recharger has replaced with a wall-mounted charger. The silly glass panes have been removed. Now, the whole thing takes up less space, and looks physically like a supply pod. A bug that causes the mech charger to be unusable has also been fixed.
- The ball gag / fluffy cuffs currently hidden between gamma and perma have to be moved. Where, you ask? To a new, tiny secret room in security, behind a poster:

🆑 Kyep
add: Gamma ship now looks like a real supply pod/ship, doesn't encourage wizards/vampires to hide in it, and has been updated with a wall-mounted charger and smarter layout.
fix: Fixed a bug causing gamma armory mech recharger to be unusable after gamma alert was declared.
/🆑

### CODE REVIEW

**Comment by FreeStylaLT** [comment]
I have noticed that the ballgag and the pink cuffs have since evolved from being in two separate locations to being in one location.. and now being in one location with a.. chair..

I don't think I'd like to see the next step in their evolution

**Comment by Regen1** [comment]
>I don't think I'd like to see the next step in their evolution

New clownspawn when

**Comment by Xantholne** [comment]
I'd like it a bit more if the table was moved so a mech recharger/mech recharge console could fit in for the durand. or even have it vertical should work still

**Comment by Fruerlund** [comment]
Love to have a door into the armory too. But that may bring access problems. Otherwise awesome work!

**Comment by Kyep** [comment]
The new secret room is hidden behind a poster that's covering up a non-reinforced (ie: easily weldable) wall. I put the chair in there for flavor. If anything would spawn there, I would imagine it to be a vengeful tunnel clown, driven insane by being sealed up in there and left to die by a previous shift.

I axed the mech recharger/console as, in all the times I've seen gamma called, I have never seen it used. When I asked other admins, they haven't seen it used either. It also doesn't make physical sense. In the original (prior to this PR) design, it relied on the room having an APC... but if you look, there is no powergrid connection. So how is that APC charged? Overall, it was a decorative gimmick that was never used and did not make physical sense. Also, it did not fit in the space of the smaller gamma pod (have to account for shuttle walls). For all of these reasons, I decided to axe it.

The door on the right side matches up with where the pod docks (and starts) at CC. If I made the gamma pod on the cyberiad accessible through the armory north of the warden's office, the pod itself would need only one door (to the east).

**Comment by FalseIncarnate** [comment]
Insane buff to the security of gamma (ironic, i know) because you can't cut through shuttle walls. If that door gets rendered inoperable, it's gonna be a pain to get in there.

Additionally, the double walls of the shuttle against station make those reinforced walls somewhat unnecessary. Perhaps change some to windows so you can see if the shuttle is there or if some cheeky engineer has set up a fort in the space to mock security from.

Frankly, magic APC aside, this feels a lot emptier than the old gamma without the mech charger. That eastern door just looks dumb when deployed since it is just a door into a wall. If you could make it actually connect to the armory or remove it, I think it will look a little better.

**Comment by SamHPurp** [comment]
Gamma was my favourite location to move my office into while clown. :(

That said I think this is a good change, and I disagree with @FalseIncarnate and others who suggest removing that door. That door is useful as it means that other stations can have Gamma facing another way. Only thing is, is that there should be some windows on the Gamma shuttle, as another way in.

**Comment by FalseIncarnate** [comment]
@SamHPurp other maps would only be able to use that if they use the same z2 map. Metastation doesn't, and future maps may make their own centcomm and thus their own gamma armory as well. I believe that door only exists for when it is "docked" at centcomm, which could just as easily be left out or adjusted (I can't think of any times someone on centcomm would need access to gamma and not already be capable of accessing it via admin commands like Jump to Turf).

**Comment by Kyep** [comment]
@FreeStylaLT @FalseIncarnate 
- A second entry door has been added to the east (armory) side. You can now get in from either side.
- Some windows have been added to the west (perma0 side. You can now tell if the gamma pod/ship is there or not simply by looking through the windows. In the event that both doors are broken, you can also break in through the windows.
- The mech charger has been re-added. I've also added in a fix which prevents it from breaking when the gamma ship is moved.

**Comment by SamHPurp** [comment]
@FalseIncarnate Except we're moving towards standardising these (hopefully). I have a PR up that standardises all Z's except for station and Centcomm... and the centcomm is only not there currently because I'm struggling with getting the shuttle rotation code to work properly... or that'd be on there too!

**Comment by Fox-McCloud** [comment]
conflicts.

File: code/game/mecha/mech_bay.dm [hunk 1]
```diff
@@ -33,6 +33,9 @@
 	component_parts += new /obj/item/weapon/stock_parts/capacitor(null)
 	component_parts += new /obj/item/stack/cable_coil(null, 1)
 	RefreshParts()
+	update_recharge_turf()
```
Comment (Fox-McCloud) [inline code/game/mecha/mech_bay.dm]: this is going to be needed to be added to the upgraded one as well.

Comment (Kyep) [inline code/game/mecha/mech_bay.dm]: Done.

**Comment by Kyep** [comment]
@Fox-McCloud Conflicts fixed.

**Pull request merged**